thrift/arg2/kv_iterator.go

```diff
@@ -5,20 +5,21 @@
 package arg2
 
 import (
-	"encoding/binary"
-	"fmt"
 	"io"
+
+	"github.com/uber/tchannel-go/typed"
 )
 
 // KeyValIterator is a iterator for reading tchannel-thrift Arg2 Scheme,
 // which has key/value pairs (k~2 v~2).
 // NOTE: to be optimized for performance, we try to limit the allocation
 // done in the process of iteration.
 type KeyValIterator struct {
-	arg2Payload           []byte
-	leftPairCount         int
-	keyOffset             int
-	valueOffset, valueLen int
+	arg2Len       int
+	rbuf          *typed.ReadBuffer
```

Comment:
i think the kv iterator isn't included in the current relay benchmark, can you do a quick benchmark of just the KeyValueIterator with the test frames to see if there's any performance impact to using the `*typed.ReadBuffer` here?

Reply:
turns out we do get 1 alloc with this:
`dev`:
```
BenchmarkKeyValIterator-8   	16034449	        73.0 ns/op	       0 B/op	       0 allocs/op
```
this branch:
```
BenchmarkKeyValIterator-8   	10106662	       116 ns/op	      64 B/op	       1 allocs/op
```
which makes sense, since we need to hold on to a `*typed.ReadBuffer` in the iterator. Will revert the change for the iterator and keep the change in `lazyCallReq`